src/React/SyntheticEvent.purs

```diff
@@ -193,3 +226,180 @@ type SyntheticWheelEvent' r
+
+targetTouches :: forall eff r. SyntheticEvent_ (SyntheticTouchEvent' r) -> Eff eff NativeTouchList
+targetTouches = get (SProxy :: SProxy "targetTouches")
+
+touches :: forall eff r. SyntheticEvent_ (SyntheticTouchEvent' r) -> Eff eff NativeTouchList
+touches = get (SProxy :: SProxy "touches")
+
+altKey :: forall eff r. SyntheticEvent_ (SyntheticKeyboardMouseTouchEvent' r) -> Eff eff Boolean
+altKey = get (SProxy :: SProxy "altKey")
+
+ctrlKey :: forall eff r. SyntheticEvent_ (SyntheticKeyboardMouseTouchEvent' r) -> Eff eff Boolean
+ctrlKey = get (SProxy :: SProxy "ctrlKey")
+
+metaKey :: forall eff r. SyntheticEvent_ (SyntheticKeyboardMouseTouchEvent' r) -> Eff eff Boolean
+metaKey = get (SProxy :: SProxy "metaKey")
+
+shiftKey :: forall eff r. SyntheticEvent_ (SyntheticKeyboardMouseTouchEvent' r) -> Eff eff Boolean
+shiftKey = get (SProxy :: SProxy "shiftKey")
+
+propertyName :: forall eff r. SyntheticEvent_ (SyntheticTransitionEvent' r) -> Eff eff String
+propertyName = get (SProxy :: SProxy "propertyName")
+
+pseudoElement :: forall eff r. SyntheticEvent_ (SyntheticAnimationTransitionEvent' r) -> Eff eff String
+pseudoElement = get (SProxy :: SProxy "pseudoElement")
+
+elapsedTime :: forall eff r. SyntheticEvent_ (SyntheticAnimationTransitionEvent' r) -> Eff eff Number
+elapsedTime = get (SProxy :: SProxy "elapsedTime")
+
+detail :: forall eff r. SyntheticEvent_ (SyntheticUIEvent' r) -> Eff eff Number
+detail = get (SProxy :: SProxy "detail")
+
+view :: forall eff r. SyntheticEvent_ (SyntheticUIEvent' r) -> Eff eff NativeAbstractView
+view = get (SProxy :: SProxy "view")
+
+deltaMode :: forall eff r. SyntheticEvent_ (SyntheticWheelEvent' r) -> Eff eff Number
+deltaMode = get (SProxy :: SProxy "deltaMode")
+
+deltaX :: forall eff r. SyntheticEvent_ (SyntheticWheelEvent' r) -> Eff eff Number
+deltaX = get (SProxy :: SProxy "deltaX")
+
+deltaY :: forall eff r. SyntheticEvent_ (SyntheticWheelEvent' r) -> Eff eff Number
+deltaY = get (SProxy :: SProxy "deltaY")
+
+deltaZ :: forall eff r. SyntheticEvent_ (SyntheticWheelEvent' r) -> Eff eff Number
+deltaZ = get (SProxy :: SProxy "deltaZ")
+
+foreign import preventDefault :: forall eff r. SyntheticEvent_ (SyntheticEvent' r) -> Eff eff Unit
+
+foreign import isDefaultPrevented :: forall eff r. SyntheticEvent_ (SyntheticEvent' r) -> Eff eff Boolean
+
+foreign import stopPropagation :: forall eff r. SyntheticEvent_ (SyntheticEvent' r) -> Eff eff Unit
+
+foreign import isPropagationStopped :: forall eff r. SyntheticEvent_ (SyntheticEvent' r) -> Eff eff Boolean
+
+foreign import persist :: forall eff r. SyntheticEvent_ (SyntheticEvent' r) -> Eff eff Unit
+
+foreign import getModifierState :: forall eff r. String -> SyntheticEvent_ (SyntheticKeyboardMouseTouchEvent' r) -> Eff eff Boolean
+
+get
+  :: forall eff l r s a
+   . RowCons l a r s
+  => IsSymbol l
+  => SProxy l
+  -> SyntheticEvent_ s
+  -> Eff eff a
+get l r = unsafeGet (reflectSymbol l) r
```

Comment:
This carries a fully polymorphic effect variable. Should it have a label?

Reply:
Are you suggesting we added in something `Eff (sythenticEvent :: SYNTHETIC_EVENT) a`? That works for me.

Reply:
Possibly. Seems like overkill to have another label for react stuff, but I'm not sure what the solution is short of consolidating the other labels we have. With 0.12 `Effect` transition coming up I wonder if it even matters.

Reply:
I am leaning toward not adding an effect label, unless we decide it is necessary. Agreed that with the 0.12 changes, it may not be worthwhile adding here. But I am open to ideas.